### PR TITLE
docs: improve debuginfo install description (fix issue #1943)

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
- **docs: improve debuginfo symbols description**
Fix issue #1943 

#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [X] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Required backport PRs have been created
- [x] Correct milestone is set

### CODE REVIEW

File: docs/manuals/source/Appendix/Debugging.rst [hunk 1]
```diff
@@ -168,16 +168,38 @@ Our Linux packages are stripped of debugging symbols, so you need an extra step
 
 
    .. code-block:: shell-session
-      :caption: Installing Bareos debug symbols on (RH)EL system
+      :caption: Installing Bareos Director debug symbols on (RH)EL system
 
-      dnf --enablerepo bareos-debuginfo install bareos-director-debuginfo
+      dnf --enablerepo=bareos-debuginfo install gdb bareos-debuginfo bareos-director-debuginfo \
+      bareos-database-common-debuginfo bareos-common-debuginfo bareos-database-tools-debuginfo
```
Comment (sebsura) [inline docs/manuals/source/Appendix/Debugging.rst:174]: This seems to mostly work except that somehow it cannot find the files themselves.  For debug traces this is good enough, but we should still look into why the files are missing.  E.g.
```
(gdb) bt
#0  0x00007f17e90797fd in ReadStateFile (dir=<optimized out>, progname=<optimized out>, port=<optimized out>) at ../../../../core/src/lib/bsys.cc:573
#1  0x000055d3d46385e0 in main (argc=2, argv=0x7ffdf3cf2468) at ../../../../core/src/dird/dird.cc:306
(gdb) info locals
filename = "/var/lib/bareos/bareos-dir.9101.state"
secure_erase_guard = {filename = "/var/lib/bareos/bareos-dir.9101.state", cleanup = true}
hdr = {id = '\000' <repeats 13 times>, version = 0, last_jobs_addr = 0, end_of_recent_job_results_list = 0, reserved = {0 <repeats 19 times>}}
file = {<std::basic_istream<char, std::char_traits<char> >> = {<std::basic_ios<char, std::char_traits<char> >> = {<std::ios_base> = {_vptr.ios_base = 0x7f17e8b8b998 <vtable for std::basic_ifstream<char, std::char_traits<char> >+64>, static boolalpha = std::_S_boolalpha, static dec = std::_S_dec, static fixed = std::_S_fixed, static hex = std::_S_hex, static internal = std::_S_internal, static left = std::_S_left, 
        static oct = std::_S_oct, static right = std::_S_right, static scientific = std::_S_scientific, static showbase = std::_S_showbase, static showpoint = std::_S_showpoint, static showpos = std::_S_showpos, static skipws = std::_S_skipws, static unitbuf = std::_S_unitbuf, static uppercase = std::_S_uppercase, static adjustfield = std::_S_adjustfield, static basefield = std::_S_basefield, 
        static floatfield = std::_S_floatfield, static badbit = std::_S_badbit, static eofbit = std::_S_eofbit, static failbit = std::_S_failbit, static goodbit = std::_S_goodbit, static app = std::_S_app, static ate = std::_S_ate, static binary = std::_S_bin, static in = std::_S_in, static out = std::_S_out, static trunc = std::_S_trunc, static beg = std::_S_beg, static cur = std::_S_cur, static end = std::_S_end, 
        _M_precision = 6, _M_width = 0, _M_flags = 4098, _M_exception = 5, _M_streambuf_state = std::_S_failbit, _M_callbacks = 0x0, _M_word_zero = {_M_pword = 0x0, _M_iword = 0}, _M_local_word = {{_M_pword = 0x0, _M_iword = 0}, {_M_pword = 0x0, _M_iword = 0}, {_M_pword = 0x0, _M_iword = 0}, {_M_pword = 0x0, _M_iword = 0}, {_M_pword = 0x0, _M_iword = 0}, {_M_pword = 0x0, _M_iword = 0}, {_M_pword = 0x0, _M_iword = 0}, {
            _M_pword = 0x0, _M_iword = 0}}, _M_word_size = 8, _M_word = 0x7ffdf3cf1a70, _M_ios_locale = {static none = 0, static ctype = 1, static numeric = 2, static collate = 4, static time = 8, static monetary = 16, static messages = 32, static all = 63, _M_impl = 0x7f17e8b93c80 <(anonymous namespace)::c_locale_impl>, static _S_classic = 0x7f17e8b93c80 <(anonymous namespace)::c_locale_impl>, 
          static _S_global = 0x7f17e8b93c80 <(anonymous namespace)::c_locale_impl>, static _S_categories = 0x7f17e8b86460 <__gnu_cxx::category_names>, static _S_once = 2, static _S_twinned_facets = 0x7f17e8b87c00 <std::locale::_S_twinned_facets>}}, _M_tie = 0x0, _M_fill = 0 '\000', _M_fill_init = false, l_M_streambuf = 0x7ffdf3cf1940, _M_ctype = 0x7f17e8b936a0 <(anonymous namespace)::ctype_c>, 
      _M_num_put = 0x7f17e8b93630 <(anonymous namespace)::num_put_c>, _M_num_get = 0x7f17e8b93640 <(anonymous namespace)::num_get_c>}, _vptr.basic_istream = 0x7f17e8b8b970 <vtable for std::basic_ifstream<char, std::char_traits<char> >+24>, _M_gcount = 0}, _M_filebuf = {<std::basic_streambuf<char, std::char_traits<char> >> = {
      _vptr.basic_streambuf = 0x7f17e8b8b878 <vtable for std::basic_filebuf<char, std::char_traits<char> >+16>, _M_in_beg = 0x0, _M_in_cur = 0x0, _M_in_end = 0x0, _M_out_beg = 0x0, _M_out_cur = 0x0, _M_out_end = 0x0, _M_buf_locale = {static none = 0, static ctype = 1, static numeric = 2, static collate = 4, static time = 8, static monetary = 16, static messages = 32, static all = 63, 
        _M_impl = 0x7f17e8b93c80 <(anonymous namespace)::c_locale_impl>, static _S_classic = 0x7f17e8b93c80 <(anonymous namespace)::c_locale_impl>, static _S_global = 0x7f17e8b93c80 <(anonymous namespace)::c_locale_impl>, static _S_categories = 0x7f17e8b86460 <__gnu_cxx::category_names>, static _S_once = 2, static _S_twinned_facets = 0x7f17e8b87c00 <std::locale::_S_twinned_facets>}}, _M_lock = {__data = {__lock = 0, 
        __count = 0, __owner = 0, __nusers = 0, __kind = 0, __spins = 0, __elision = 0, __list = {__prev = 0x0, __next = 0x0}}, __size = '\000' <repeats 39 times>, __align = 0}, _M_file = {_M_cfile = 0x0, _M_cfile_created = false}, _M_mode = 0, _M_state_beg = {__count = 0, __value = {__wch = 0, __wchb = "\000\000\000"}}, _M_state_cur = {__count = 0, __value = {__wch = 0, __wchb = "\000\000\000"}}, _M_state_last = {
      __count = 0, __value = {__wch = 0, __wchb = "\000\000\000"}}, _M_buf = 0x0, _M_buf_size = 8192, _M_buf_allocated = false, _M_reading = false, _M_writing = false, _M_pback = 0 '\000', _M_pback_cur_save = 0x0, _M_pback_end_save = 0x0, _M_pback_init = false, _M_codecvt = 0x7f17e8b93610 <(anonymous namespace)::codecvt_c>, _M_ext_buf = 0x0, _M_ext_buf_size = 0, _M_ext_next = 0x0, _M_ext_end = 0x0}}
(gdb) list
568	in ../../../../core/src/lib/bsys.cc
```

File: docs/manuals/source/Appendix/Debugging.rst [hunk 2]
```diff
@@ -168,16 +168,38 @@ Our Linux packages are stripped of debugging symbols, so you need an extra step
...
       # Enable and activate bareos-debuginfo repository
       zypper modifyrepo --enable --refresh --gpgcheck bareos-debuginfo
 
-      zypper install bareos-director-debuginfo
+      zypper install gdb bareos-debuginfo bareos-director-debuginfo bareos-common-debuginfo \
+      bareos-database-common-debuginfo bareos-database-tools-debuginfo
+
+.. note::
+   For rpm: you maybe want to install all corresponding **-debuginfo** of installed **bareos-**
+   packages if you want to debug all.
```
Comment (sebsura) [inline docs/manuals/source/Appendix/Debugging.rst:187]: Maybe we want to mention that you can do so with
```
dnf debuginfo-install bareos-*-debuginfo
```
zypper probably works similarly.

**Review by sebsura — CHANGES_REQUESTED** [review]
The changes are great.  I added some comments to the documentation that you added.

File: docs/manuals/source/Appendix/Debugging.rst (hunk 2 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/Appendix/Debugging.rst:187]: Nice improvement for dnf call I will use that one.
and thank to force myself to go further, zypper has something similar :-)
```
zypper -v --plus-content bareos-debuginfo install -r bareos-debuginfo bareos-*-debuginfo
```
will use that too.

Comment (bruno-at-bareos) [reply docs/manuals/source/Appendix/Debugging.rst]: done

**Comment by bruno-at-bareos** [comment]
@sebsura as discussed, I've redone the section, by putting in first place how to install all, then only describe specific daemon with fd as an example.
To make the commits more organized, I've done a rebase, sorry if that force you for a full review.

File: docs/manuals/source/Appendix/Debugging.rst [hunk 3]
```diff
@@ -158,26 +158,60 @@ Installing debug symbols packages
 .. index::
    single: debug symbols package; dbg; debuginfo; debug
 
-Our Linux packages are stripped of debugging symbols, so you need an extra step to install their rpm debuginfo or deb dbg equivalent
+Our Linux packages are stripped of debugging symbols, so you need an extra step to install their rpm debuginfo or deb dbg equivalent.
```
Comment (sebsura) [inline docs/manuals/source/Appendix/Debugging.rst:161]:
```suggestion
Our binaries do not contain debug symbols, but as they are needed for proper debugging, we package the debug symbols separately.
```

Those symbols are almost always called `debug symbols` (at least i have never heard `debugging symbols` before).  Im not sure about the other changes, what do you think ?

File: docs/manuals/source/Appendix/Debugging.rst [hunk 4]
```diff
@@ -158,26 +158,60 @@ Installing debug symbols packages
...
+
+
+.. code-block:: shell-session
+   :caption: Installing all Bareos debug symbols on (open)SUSE system
+
+   zypper --plus-content bareos-debuginfo --plus-content bareos-debugsource install bareos*debuginfo bareos*debugsource
+
+
+
+.. Note:: If you want to debug only a specific daemon, you have to install only its related **-debuginfo** packages. For example with |fd|
```
Comment (sebsura) [inline docs/manuals/source/Appendix/Debugging.rst:182]:
```suggestion
.. Note:: If you want to debug only a specific daemon, you only need to install its related **-debuginfo** packages. For example with |fd|
```

File: docs/manuals/source/Appendix/Debugging.rst [hunk 5]
```diff
@@ -158,26 +158,60 @@ Installing debug symbols packages
...
+   :caption: Installing |fd| debug symbols package on deb system
 
-      dnf --enablerepo bareos-debuginfo install bareos-director-debuginfo
+   apt install bareos-dbg gdb
+
+
+.. code-block:: shell-session
+   :caption: Installing |fd| debug symbols on (RH)EL system
+
+   dnf debuginfo-install gdb bareos-debuginfo bareos-filedaemon-debuginfo bareos-filedaemon-python3-plugin-debuginfo bareos-common-debuginfo bareos-debugsource
```
Comment (sebsura) [inline docs/manuals/source/Appendix/Debugging.rst:193]: maybe we also want to use `bareos-filedaemon-*-debuginfo` here ?

**Review by sebsura — CHANGES_REQUESTED** [review]

File: docs/manuals/source/Appendix/Debugging.rst (hunk 3 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/Appendix/Debugging.rst:161]: done

File: docs/manuals/source/Appendix/Debugging.rst (hunk 5 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/Appendix/Debugging.rst:193]: good catch, done

File: docs/manuals/source/Appendix/Debugging.rst (hunk 4 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/Appendix/Debugging.rst]: sorry this is now in.

**Review by sebsura — APPROVED** [review]
Great changes. Thanks for your work!

**Comment by bruno-at-bareos** [comment]
Backport created now last build for merge

**Pull request merged**